### DISCUSSION TITLE
Get FileName and FilePath without actually reading in the file?

### DISCUSSION THREAD

**Discussion by rjkroll** [nextgenhealthcare/connect#5671] [Q&A] [answered]
Hi,

Using a File Reader as a source and pointing to large CSV's causing the channel to read the file into memory and exceed heap space sizes.

Normally, I need to get the FileName and FilePath from File Readers for user in downstream processes by using the following in a transformer step.   But, with heap space issues on these larger files, is this still possible to do?  Can you get the name and path still without having to read the whole file into memory like it currently does?   Or do I have to use Javascript reader source and use some kind of javascript?  

var fname = sourceMap.get('originalFilename')
var path =  sourceMap.get('fileDirectory')

I essentially was using these two values to pass into a powershell script for the location of a CSV to import for normally sized CSV's (via a File Reader).   All I need essentially is some kind of Mirth channel that polls a folder and spits back filepath+filenames without actually doing anything to the files.   Maybe im looking at this the wrong way?

**Comment by jonbartels** [comment, marked as answer]
I like this question. If Mirth were a human on the console you want it to do an `ls -l` to show the file name and size before doing a `cat`.

To the best of my knowledge there is no way to do this in the source file reader.

Are these local files or remote files (FTP/SFTP)? Local is a LOT easier to implement a workaround for

Workarounds:
1. Use attachments - you still read the file, but it is read as an attachment instead of as message content. This still reads the big message but it is more efficiently stored in Mirth https://forums.mirthproject.io/articles/175046-using-attachments
2. Use a JavascriptReader and call the file list operations but don't read the whole content
3. Use a syscall to execute calls as needed - see https://github.com/nextgenhealthcare/connect-examples/tree/master/Code%20Templates/Execute%20Runtime%20Command . *Since you're already calling powershell this may be the best option*

**Reply by rjkroll** [reply]
Updating for a solution:   Thanks for the pointers, this lead me to use a Javascript reader that just does the following to return just the names of the files in the folder without reading them in and exceeding heap space

```
var folderPath = "E:/pathToFiles/";
var folder = new Packages.java.io.File(folderPath);
var fileList = folder.listFiles().filter(function(file) {
  return file.isFile();
}).map(function(file) {
  return file.getName();
});

for (var i = 0; i < fileList.length; i++) {
  var fileName = fileList[i];
 logger.error(fileName)
}
```
